packages/card/src/vaadin-card.js

```diff
@@ -31,11 +57,233 @@ class Card extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
   static get styles() {
     return css`
       :host {
+        display: flex;
+        flex-direction: column;
+        padding: var(--_padding);
+        gap: var(--_gap);
+        --_padding: var(--vaadin-card-padding, 1em);
+        --_gap: var(--vaadin-card-gap, 1em);
+        --_media: 0;
+        --_title: 0;
+        --_subtitle: 0;
+        --_header: max(var(--_header-prefix), var(--_title), var(--_subtitle), var(--_header-suffix));
+        --_header-prefix: 0;
+        --_header-suffix: 0;
+        --_content: 0;
+        --_footer: 0;
+      }
+
+      :host(:not([theme~='horizontal'])) {
+        justify-content: space-between;
+      }
+
+      :host(:is(:has([slot='media']), [has-media])) {
+        --_media: 1;
+      }
+
+      :host(:is(:has([slot='title']), [has-title])) {
+        --_title: 1;
+      }
+
+      :host(:is(:has([slot='subtitle']), [has-subtitle])) {
+        --_subtitle: 1;
+      }
+
+      :host(:is(:has([slot='header']), [has-header])) {
+        --_header: 1;
+        --_title: 0;
+        --_subtitle: 0;
+      }
+
+      :host(:is(:has([slot='header-prefix']), [has-header-prefix])) {
+        --_header-prefix: 1;
+      }
+
+      :host(:is(:has([slot='header-suffix']), [has-header-suffix])) {
+        --_header-suffix: 1;
+      }
+
+      :host(:is(:has(> :not([slot])), [has-content])) {
+        --_content: 1;
+      }
```

Comment:
Looks like this doesn't support plain text as card content, e.g. the following works:

```html
<vaadin-card>
  <div>Content</div>
</vaadin-card>
```

But the following does not work (the content isn't visible):

```html
<vaadin-card>Content</vaadin-card>
```